File: front/lib/features/clases/models/actualizar_horarios_request.dart
// lib/features/clases/models/actualizar_horarios_request.dart
class ActualizarHorariosRequest {
  final int alumnoId;
  final int claseId;
  final List<Map<String, String>> horarios;

  ActualizarHorariosRequest({
    required this.alumnoId,
    required this.claseId,
    required this.horarios,
  });

  Map<String, dynamic> toJson() => {
        'alumno_id': alumnoId,
        'clase_id': claseId,
        'horarios': horarios,
      };
}

File: back/src/models/actualizar_horarios_request.py
# src/models/actualizar_horarios_request.py
from pydantic import BaseModel
from typing import List

class HorarioUpdate(BaseModel):
    dia: str
    hora: str
    aula: str

class ActualizarHorariosRequest(BaseModel):
    alumno_id: int
    clase_id: int
    horarios: List[HorarioUpdate]